Source/JavaScriptCore/parser/Lexer.cpp

```diff
@@ -1935,27 +1935,30 @@ JSTokenType Lexer<T>::lexWithoutClearingLineTerminator(JSToken* tokenRecord, Opt
     ASSERT(currentOffset() >= currentLineStartOffset());
     tokenRecord->m_startPosition = currentPosition();
 
+    LChar type = m_current;
+
     if (atEnd()) {
         token = EOFTOK;
         goto returnToken;
     }
 
-    CharacterType type;
-    if (isLatin1(m_current)) [[likely]]
-        type = static_cast<CharacterType>(typesOfLatin1Characters[m_current]);
-    else {
-        char32_t codePoint;
-        U16_GET(m_code, 0, 0, m_codeEnd - m_code, codePoint);
-        if (isNonLatin1IdentStart(codePoint))
-            type = CharacterNonLatin1IdentifierStart;
-        else if (isLineTerminator(m_current))
-            type = CharacterLineTerminator;
-        else
-            type = CharacterInvalid;
+    if constexpr (sizeof(T) != sizeof(LChar)) {
+        if (!isLatin1(m_current)) [[unlikely]] {
+            char32_t codePoint;
+            U16_GET(m_code, 0, 0, m_codeEnd - m_code, codePoint);
+            if (isNonLatin1IdentStart(codePoint)) {
+                // We are hijacking white space characters for non-latin1 identifier start in the following dispatch since we will never see
+                // these characters in the switch because of `skipWhitespace()` call.
+                type = ' ';
+            } else if (isLineTerminator(m_current))
+                type = '\n';
+            else
+                type = '\0';
+        }
     }
 
     switch (type) {
-    case CharacterGreater:
+    case  62 /* > CharacterGreater */: {
```

Comment:
Nit: Comment might be clearer as `/* 62 = '>' CharacterGreater */`. Ditto for the others below.